test/Microsoft.ML.Functional.Tests/Validation.cs

```diff
@@ -138,5 +139,27 @@ public void TrainWithValidationSet()
             Common.AssertMetrics(trainMetrics);
             Common.AssertMetrics(validMetrics);
         }
+
+        /// <summary>
+        /// Test cross validation R^2 metric to return NaN when given fewer data
+        /// than needed to infer metric calculation. R^2 is NaN when given folds
+        /// with less than 2 rows of training data.
+        /// </summary>
+        [Fact]
+        public void TestCrossValidationResultsWithNotEnoughData()
+        {
+            var mlContext = new MLContext(1);
+            // Get data and set up sample regression pipeline.
+            var data = mlContext.Data.LoadFromTextFile<Iris>(TestCommon.GetDataPath(DataDir, TestDatasets.iris.trainFilename), hasHeader: true);
+            var dataFirstTenRows = mlContext.Data.TakeRows(data, 10);
+            var pipeline = mlContext.Transforms.Concatenate("Features", Iris.Features)
+                .Append(mlContext.Regression.Trainers.OnlineGradientDescent());
+
+            // Check that NaN is returned with fold given less than 2 rows of training data.
+            // With dataset of 10 rows, number of folds will be 6.
+            var cvResults = mlContext.Regression.CrossValidate(dataFirstTenRows, pipeline, numberOfFolds: 6);
```

Comment:
Could use a normal training, then evaluate on one row of data. This way you could avoid the randomness of creating the CV folds (which is current curtailed using a seed).

One secondary test would be testing zero rows in the scoring dataset.

Reply:
Edited unit test per the above, and added secondary test for the 0 row scenario. Thanks!